src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -298,7 +343,7 @@ private ResourceContext GetResourceContext(object graph, ODataSerializerContext
         }
 
         private void WriteDeltaComplexAndExpandedNavigationProperty(IEdmProperty edmProperty, SelectExpandClause selectExpandClause,
-            ResourceContext resourceContext, ODataWriter writer)
+            ResourceContext resourceContext, ODataWriter writer, Type navigationPropertyType = null)
```

Comment:
Curious, cant's we extract navigation property type from `edmProperty` parameter?

Reply:
Will it be of type `Type`. How easy will it be to extract it from `edmProperty`.